gpii/node_modules/settingsHandlers/src/WebSocketsComponent.js

```diff
@@ -0,0 +1,262 @@
+        modelListeners: {
+            "clients.*": {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.clientsChanged",
+                args: "{change}"
+            },
+            "settings.*": {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.settingsChanged",
+                args: "{change}"
+            }
+        },
+        invokers: {
+            addClient: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.addClient",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            },
+            removeClient: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.removeClient",
+                args: ["{that}", "{arguments}.0"]
+            },
+            getSettingsForId: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.getSettingsForId",
+                args: ["{that}", "{arguments}.0"]
+            },
+            setSettingsForId: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettingsForId",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            },
+            //getSettings: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.getSettings",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            //setSettingsIndividually: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettingsIndividually",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            //setSettings: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettings",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            get: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.get",
+                args: ["{arguments}.0", "{that}"]
+            },
+            set: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.set",
+                args: ["{arguments}.0", "{that}"]
+            }
+        }
+    });
+
+    ///////////////// Clients /////////////////////
+
+    gpii.settingsHandlers.WebSocketsComponent.clientsChanged = function (a) {
+        var util = require("util");
+        console.log("A change in clients has been registered: " + util.inspect(a));
+    }
+
+    gpii.settingsHandlers.WebSocketsComponent.addClient = function (that, solutionId, client) {
+       var solution = solutionId.replace(/\./g, "\\.");
+       var type = fluid.get(that.model.clients, [solutionId, client.id]) === undefined ? "ADD": null;
+
+       // How can I set these options to the applier in a declarative way?
+       //
+       that.applier.options.resolverGetConfig = fluid.model.escapedGetConfig;
+       that.applier.options.resolverSetConfig = fluid.model.escapedSetConfig;
+
+       that.applier.change("clients." + solution + "." + client.id, client, type);
```

Comment:
Antranig, this is something I experienced since the beginning.
When I use __that.applier.change(["clients", solutionId, client.id], client, type);__ I get the following error:
```
10:44:36.881:  FATAL ERROR: Uncaught exception: Object clients,org.chrome.cloud4chrome,sAnJo4iPLfgjpdUFZEhB has no method 'charAt'
TypeError: Object clients,org.chrome.cloud4chrome,sAnJo4iPLfgjpdUFZEhB has no method 'charAt'
```